crates/starknet-devnet-core/src/state/mod.rs

```diff
@@ -3,23 +3,79 @@ use std::collections::HashMap;
 use blockifier::state::cached_state::{
     CachedState, GlobalContractCache, GLOBAL_CONTRACT_CACHE_SIZE_FOR_TEST,
 };
-use blockifier::state::state_api::StateReader;
-use starknet_rs_ff::FieldElement;
+use blockifier::state::state_api::{State, StateReader};
+use starknet_api::core::CompiledClassHash;
+use starknet_api::hash::StarkFelt;
+use starknet_types::constants::MAX_BYTECODE_SIZE_LIMIT;
 use starknet_types::contract_address::ContractAddress;
 use starknet_types::contract_class::ContractClass;
-use starknet_types::contract_storage_key::ContractStorageKey;
-use starknet_types::felt::{ClassHash, CompiledClassHash, Felt};
+use starknet_types::felt::{ClassHash, Felt};
 
 use self::state_diff::StateDiff;
-use crate::error::{DevnetResult, Error, StateError};
-use crate::traits::{DevnetStateReader, StateChanger, StateExtractor};
+use self::state_readers::DictState;
+use crate::error::{DevnetResult, Error};
 
 pub(crate) mod state_diff;
+mod state_readers;
 pub mod state_update;
 
+pub trait CustomStateReader {
```

Comment:
can those traits be moved to traits.rs

Reply:
They could, but honestly, having a dedicated traits.rs file for storing all traits looks like an antipattern to me. Maybe it would be more meaningful if it was in `state_readers.rs`.